### ISSUE TITLE
Interaction of "add" with "flow" / "flow-resolvers" plugin

### ISSUE THREAD

**Issue by xanf** [dotansimha/graphql-code-generator#1117]
**Is your feature request related to a problem? Please describe.**
I'm using `add` plugin with `flow` and `flow-resolvers` plugin and having funny issue.
I'm using custom context type and also want to add `/* eslint-disable */` to the top of my code to keep my CI and commit hooks happy (I know about `.eslintignore` file but I assume the problem is broader). My config is: 
```
    plugins:
      - add: "/* eslint-disable */"
      - add: 'import { type ContextT } from "../gql";'
      - flow
      - flow-resolvers
```
Right now, lines added by "add" plugin are added directly to the top of file resulting in following output:
```
/* eslint-disable */
import { type ContextT } from "../gql";
/* @flow */

...
```
which makes flow unhappy - it wants flow pragma at the very beginning of file. right now I'm working around this issue with:
```
    plugins:
      - add: "// @flow"
      - add: "/* eslint-disable */"
      - add: 'import { type ContextT } from "../gql";'
      - flow
      - flow-resolvers
```
which is pretty ugly but works :)
```js
// @flow
/* eslint-disable */
import { type ContextT } from "../gql";
/* @flow */
```

**Describe the solution you'd like**
I didn't dig too deeply into interaction of plugins, but we need to ensure that flow pragma will be added to the very start of the file, or have option in `flow` / `flow-resolvers` not to add flow pragma to the start of the file (which *might* be useful for that insane people who configured flow to check all files not just with pragmas)

Before hacking around the code I would like to hear you input on this

**Comment by dotansimha** [comment]
Hi @xanf ! 
Thanks for reporting it. 
I think we need to add to the plugin api a `prefix` option, and it will append the context as prefix before all other plugins. This way the `flow` plugin can declare it's own prefix (`// @flow`) and it will be added before other `add` plugins.

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/1869

**Comment by dotansimha** [comment]
Fixed in `1.2.0`.
